lib/remote_persistent_term/fetcher/http/cache.ex

```diff
@@ -0,0 +1,54 @@
+defmodule RemotePersistentTerm.Fetcher.Http.Cache do
+  @moduledoc """
+  A module to assist in following the HTTP Caching spec.
+
+  See more: https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching
+  """
+
+  @max_age_regex ~r/max-age=(\d+)/
+  @default_age 0
+
+  @doc """
+  Calculates the refresh interval from the cache-control and age headers.
+  Caching directives are case-insensitive. Multiple directives are permitted and must be comma-separated
+
+  Refresh interval is in milliseconds
+  """
+  @spec refresh_interval(Req.Response.t()) :: {:ok, pos_integer()} | {:error, term()}
```

Comment:
```suggestion
  @spec refresh_interval(Req.Response.t()) :: {:ok, pos_integer()} | {:error, reason :: String.t()}
```
Looks like you have all of the errors handled and always return a nice message string